### PR TITLE
Make marshaled Vendor ID SAN in certs the same as in nuts-crypto

### PR DESCRIPTION
**Pull request by reinkrul**
Also did some cleanup.

PR is based on https://github.com/nuts-foundation/nuts-discovery/pull/58 and https://github.com/nuts-foundation/nuts-discovery/pull/57 (so those should be merged first).

### CODE REVIEW

**Comment by woutslakhorst** [comment]
@reinkrul could you rebase on master?

File: src/test/kotlin/nl/nuts/discovery/service/CertificatesApiServiceImplTest.kt [hunk 1]
```diff
@@ -107,6 +106,7 @@ class CertificatesApiServiceImplTest {
 
         nutsCertificateRequestRepository.save(req)
         val x509 = certificatesApiServiceImpl.sign(req)
+        println( Certificate.fromX509Certificate(x509, "", "").toPem())
```
Comment (woutslakhorst) [inline src/test/kotlin/nl/nuts/discovery/service/CertificatesApiServiceImplTest.kt:109]: left-over debugging?

File: src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package nl.nuts.discovery.store.entity
+
+import org.bouncycastle.asn1.x509.OtherName
+import javax.persistence.AttributeConverter
+import javax.persistence.Convert
+
+
+data class PartyId(val oid: String, val value: String) {
```
Comment (woutslakhorst) [inline src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt:8]: Missing PartyIDTest.kt

Comment (woutslakhorst) [reply src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt:8]: Missing doc on class

File: src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    companion object {
+        fun parse(fullyQualified: String): PartyId {
+            val parts = fullyQualified.split(":")
+            if (parts.size != 4) {
+                throw IllegalArgumentException("Invalid number of parts in OID URN: $fullyQualified")
+            }
+            if (parts[0] != "urn" || parts[1] != "oid") {
```
Comment (woutslakhorst) [inline src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt:16]: requiring `oid` is very restrictive. We have to remember this.....

File: src/main/kotlin/nl/nuts/discovery/store/entity/Certificate.kt [hunk 4]
```diff
@@ -49,28 +44,26 @@ class Certificate {
...
-                val generalName = it as List<*> // GeneralName
-                if (generalName[0] == GeneralName.otherName) {
-                    // then this should be a DLSequence
-                    val seq = generalName[1] as DLSequence
-                    if (seq.getObjectAt(0) == NutsCertificateRequest.NUTS_VENDOR_EXTENSION) {
-                        return (seq.getObjectAt(1) as DERUTF8String).toString()
+        private fun getSinglePartyID(certificate: X509Certificate): PartyId? {
+            return certificate.subjectAlternativeNames
+                    // Take SANs of type OtherName
+                    .filter { it[0] == 0 }
```
Comment (woutslakhorst) [inline src/main/kotlin/nl/nuts/discovery/store/entity/Certificate.kt:56]: magic number

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt (hunk 3 above)
Comment (reinkrul) [reply src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt:16]: Party identifiers (subject/custodian/actor) are always in the form of `urn:oid:<oid>:<value>`, as specified (https://nuts-documentation.readthedocs.io/en/latest/pages/technical/identifiers.html). So I regard the strictness of requiring it to be in that exact format as an improvement over the current situation (free format).

File: src/main/kotlin/nl/nuts/discovery/store/entity/Certificate.kt (hunk 4 above)
Comment (reinkrul) [reply src/main/kotlin/nl/nuts/discovery/store/entity/Certificate.kt:56]: Found a sweet constant for it (`GeneralName.otherName`)

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**